### PR TITLE
[feature-wip](multi-catalog) Collect external table statistics.

### PR DESCRIPTION
**Pull request by Jibing-Li**
# Proposed changes

Issue Number: close #xxx

Collect HMS external table statistic information through external metadata.
Insert the result into __internal_schema.column_statistics using insert into SQL.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/datasource/PooledHiveMetaStoreClient.java [hunk 1]
```diff
@@ -108,6 +105,14 @@ public List<FieldSchema> getSchema(String dbName, String tblName) {
         }
     }
 
+    public List<ColumnStatisticsObj> getTableColumnStatistics(String dbName, String tblName, List<String> columns) {
+        try (CachedClient client = getClient()) {
+            return client.client.getTableColumnStatistics(dbName, tblName, columns);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/PooledHiveMetaStoreClient.java:110]: Maybe we need a cache for this statistics.
After #14134 merged

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.28 seconds
 load time: 438 seconds
 storage size: 17178578853 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221114140146_clickbench_pr_45873.html

File: fe/fe-core/src/main/java/org/apache/doris/common/util/Util.java [hunk 2]
```diff
@@ -556,4 +556,16 @@ public static void logAndThrowRuntimeException(Logger logger, String msg, Throwa
         logger.warn(msg, e);
         throw new RuntimeException(msg, e);
     }
+
+    public static long bytesToLongBigEndian(byte[] bytes) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/Util.java:560]: Nowhere to use it?

File: fe/fe-core/src/main/java/org/apache/doris/statistics/IcebergAnalysisJob.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        Table icebergTable = getIcebergTable();
+        TableScan tableScan = icebergTable.newScan().includeColumnStats();
+        HMSStatistic stats = new HMSStatistic();
+        for (FileScanTask task : tableScan.planFiles()) {
+            processDataFile(task.file(), task.spec(), stats);
+        }
+    }
+
+    private Table getIcebergTable() {
+        org.apache.iceberg.hive.HiveCatalog hiveCatalog = new org.apache.iceberg.hive.HiveCatalog();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/IcebergAnalysisJob.java:52]: Why not get table from HMSExternalTable directly?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/external/HMSExternalTable.java [hunk 4]
```diff
@@ -293,5 +315,26 @@ private void initPartitionColumns() {
...
+        PooledHiveMetaStoreClient client = ((HMSExternalCatalog) catalog).getClient();
+        return client.getPartitionColumnStatistics(dbName, name, partNames, columns);
+    }
+
+    public Partition getPartition(List<String> partitionValues) {
+        PooledHiveMetaStoreClient client = ((HMSExternalCatalog) catalog).getClient();
+        return client.getPartition(dbName, name, partitionValues);
+    }
+
+    public Table getTable() {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/external/HMSExternalTable.java:335]: We already had `remoteTable` in this class.

File: fe/fe-core/src/main/java/org/apache/doris/statistics/IcebergAnalysisJob.java (hunk 3 above)
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/IcebergAnalysisJob.java:67]: This function is to get org.apache.iceberg.Table.
getTable in HMSExternalTable is to get org.apache.hadoop.hive.metastore.api.Table.
I think we can keep the code here.

File: fe/fe-core/src/main/java/org/apache/doris/statistics/IcebergAnalysisJob.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        }
+        if (colId == -1) {
+            throw new RuntimeException(String.format("Column %s not exist.", col.getName()));
+        }
+        dataSize += dataFile.columnSizes().get(colId);
+        numRows += dataFile.recordCount();
+        numNulls += dataFile.nullValueCounts().get(colId);
+    }
+
+    private void updateStats() throws Exception {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/IcebergAnalysisJob.java:100]: This should be moved to a unified place.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**